api/src/controllers/put/administrative-sanction.js

```diff
@@ -35,6 +36,7 @@ const AdministrativeSanctionPost = require('../post/administrative-sanction');
  * @returns object containing the operation's status and created records
  */
 exports.editRecord = async function(args, res, next, incomingObj) {
+  console.log('1', incomingObj);
```

Comment:
Remove

Reply:
> 
> 
> > Quick overview of the changes:
> > `forceAnonymous` is a new field in the `issuedTo` object.
> > 
> > * It indicates if the user has manually enforced anonymity or not.
> > * It is not used to determine redactions, etc, that is all done through the read roles still
> > 
> > Documents now have their read roles updated based on the anonymity checks against a records `issuedTo` object.
> > 
> > * This is only checked for record types whose schema has `issuedTo`.
> 
> Why would we need a forceAnonymous. From what I understand, this flag is able to hide a users information on selected front ends if it is set as true. For example, on NRCED the issued to will not reveal the persons details, however, if we look in NRPTI, the persons details will remain. Why not use the existing roles and permissions paradigm we have and just tag read as [sysadmin] or [public, sysadmin] instead. That way there isn't another thing for us to manage when displaying data in our front ends or future front ends. It also allows us to be able to control according to types of users. So say, there's something we want to hide from a BCMI Admin, it can be easily done with very little code.
> 
> Looking at our current code in master, it looks like issuedTo already has read, write fields.



> 
> 
> > Quick overview of the changes:
> > `forceAnonymous` is a new field in the `issuedTo` object.
> > 
> > * It indicates if the user has manually enforced anonymity or not.
> > * It is not used to determine redactions, etc, that is all done through the read roles still
> > 
> > Documents now have their read roles updated based on the anonymity checks against a records `issuedTo` object.
> > 
> > * This is only checked for record types whose schema has `issuedTo`.
> 
> Why would we need a forceAnonymous. From what I understand, this flag is able to hide a users information on selected front ends if it is set as true. For example, on NRCED the issued to will not reveal the persons details, however, if we look in NRPTI, the persons details will remain. Why not use the existing roles and permissions paradigm we have and just tag read as [sysadmin] or [public, sysadmin] instead. That way there isn't another thing for us to manage when displaying data in our front ends or future front ends. It also allows us to be able to control according to types of users. So say, there's something we want to hide from a BCMI Admin, it can be easily done with very little code.
> 
> Looking at our current code in master, it looks like issuedTo already has read, write fields.

The `forcedAnonymous` flag is there because an admin user can choose to make a record anonymous, even if they have a birth date and are an adult.  The reason we need it, is because we need to distinguish between records that are anonymous because of the underage rules, and those that are anonymous because we've manually set them to be.

If you tried to update a record that had an adult person (valid birth date, etc), but they were manually set to anonymous, how would you know NOT to update the read roles?  Because the business logic says they should be public, but the admin wants them to remain anonymous.  So we have to store that value.  We still use the read roles to determine redactions like normal, that `forceAnonymous` flag is only there to indicate that the admin wants to disregard the business rules around anonymity, and force it to be anonymous (presumably because they have a good reason).